xrcap/rootCrownSegmentation.cpp

```diff
@@ -347,7 +347,13 @@ int segment(string grayscale_images_directory, int sampling, string binary_image
 		resize(grayscale_image, grayscale_image, Size(), scale, scale, INTER_LINEAR);
 		Mat binary_image; // thresholded image data
 
-		threshold_value = threshold(grayscale_image, binary_image, 0, 255, CV_THRESH_TRIANGLE);
+		// If the user defines a threshold value, use it for segmentation
+		if (threshold_value > -1) {
+			threshold_value = cv::threshold(grayscale_image, binary_image, user_threshold_value, 255, CV_THRESH_BINARY);
+		// Otherwise, determine a threshold value for segmentation
+		} else {
+			threshold_value = threshold(grayscale_image, binary_image, 0, 255, CV_THRESH_TRIANGLE);
+		}
```

Comment:
Thoughts on changing CV_THRESH_TRIANGLE? I want to do more research about this algorithm but if Otsu is a better fit, we can change it to CV_THRESH_OTSU.